clang/test/Driver/sycl-mlir.cpp

```diff
@@ -85,14 +222,27 @@
 // RUN: %clangxx -### --sysroot=%S/Inputs/SYCL -fsycl-device-only      \
 // RUN: -target x86_64-unknown-linux-gnu -fsycl -c -emit-mlir          \
 // RUN: -fsycl-targets=spir64-unknown-unknown-syclmlir %s              \
+// RUN: -fsycl-raise-host                                              \
 // RUN: -o foo.mlir 2>&1                                               \
-// RUN: | FileCheck -check-prefix=CHK-INVOKE-MLIR %s
+// RUN: | FileCheck -check-prefix=CHK-INVOKE-MLIR-RAISE %s
 //
-// CHK-INVOKE-MLIR-O: "{{.*}}cgeist" "-S" "{{.*}}.cpp" "-o" "foo.mlir" "--args" "-cc1"
+// CHK-INVOKE-MLIR-RAISE: "{{.*}}cgeist" "-emit-llvm" "{{.*}}sycl-mlir.cpp" "-o" "{{.*}}.bc"
+// CHK-INVOKE-MLIR-RAISE: "{{.*}}mlir-translate" "-o" "{{.*}}.mlir" "--import-llvm" "{{.*}}.bc"
+// CHK-INVOKE-MLIR-RAISE: "{{.*}}cgeist" "-S" "{{.*}}.cpp" "-sycl-use-host-module" "{{.*}}.mlir" "-o" "{{.*}}.mlir"
 
 // RUN: %clangxx -### --sysroot=%S/Inputs/SYCL -fsycl-device-only       \
 // RUN: -target x86_64-unknown-linux-gnu -fsycl -c -Xcgeist -S          \
 // RUN: -fsycl-targets=spir64-unknown-unknown-syclmlir %s 2>&1          \
 // RUN: | FileCheck -check-prefix=CHK-INVOKE-ARG-PASS %s
 //
 // CHK-INVOKE-ARG-PASS: "{{.*}}cgeist" "-emit-llvm" "-S" "{{.*}}.cpp" "-o" "{{.*}}.bc" "--args" "-cc1"
+
+// RUN: %clangxx -### --sysroot=%S/Inputs/SYCL -fsycl-device-only       \
+// RUN: -target x86_64-unknown-linux-gnu -fsycl -c -Xcgeist -S          \
+// RUN: -fsycl-raise-host                                               \
+// RUN: -fsycl-targets=spir64-unknown-unknown-syclmlir %s 2>&1          \
+// RUN: | FileCheck -check-prefix=CHK-INVOKE-ARG-PASS-RAISE %s
+
+// CHK-INVOKE-ARG-PASS-RAISE: "{{.*}}cgeist" "-emit-llvm" "-S" "{{.*}}.cpp" "-o" "{{.*}}.bc" "--args" "-cc1"
+// CHK-INVOKE-ARG-PASS-RAISE: "{{.*}}mlir-translate" "-o" "{{.*}}.mlir" "--import-llvm" "{{.*}}.bc"
```

Comment:
looks like we are using cgeist to generate a LLVM bitcode file, and use it as input to mlir-translate.
Shouldn't the host bitcode be generated using clang?

Reply:
That is the stub file being built. Do you want me to add a check for `clang` in between to make this more explicit?

Reply:
Yes, I think that would be useful.